### DISCUSSION TITLE
"line-style" for different edge types?

### DISCUSSION THREAD

**Discussion by bensonnw** [cytoscape/cytoscape.js#3175] [Q&A]
In our project, we have different types for edges, we can't use something like colour or line width to distinguish them, we look at "line-style" option, but unfortunately, Cytoscape only supports 3 "line-style"s (solid, dotted, or dashed, and we have more than 3 kinds of edge), wondering what's the best way to represent different edges base on their types? any extensions we can use to support more "line-style"?

**Comment by maxkfranz** [comment]
That's a great use case for why we support custom patterns.

**Reply by bensonnw** [reply]
@maxkfranz Any details or information on "custom patterns"? docs? thank you!

**Reply by maxkfranz** [reply]
The `line-dash-*` props: https://js.cytoscape.org/#style/edge-line

**Reply by bensonnw** [reply]
Thx @maxkfranz  for pointing it out. the `line-dash-*` seem to control the alternating lengths of lines and gaps, if using those for different edges, will the differences not be "noticeable" when showing thousands edges, especially when zooming out?

**Reply by maxkfranz** [reply]
You're right. That's generally why other properties, like colour, are primarily used for edges.
